### PR TITLE
Solar Vignette, Solar Dataset Update

### PR DESCRIPTION
**Pull request by mintzj**
Removes bats from solar PV dataset, renames solar_pv to solar_PV, adds solar vignette, adds seed to all vignettes.

### CODE REVIEW

**Comment by mintzj** [comment]
@DAPPERstats Is there a fix for the 'ylim' issue in command-line-example vignette?  I keep running into this and I am not sure what to do about it.  My guess right now is that it has something to do with namespace issues and the same name for files being used in all vignettes.   Any ideas?

The only time I could replicate this behavior was when I had the wrong dataset (wind-examples) loaded and tried to run one of the plots at the end of the command-line-example.Rmd.

``Error: processing vignette 'wind-examples.Rmd' failed with diagnostics:
need finite 'ylim' values``

**Comment by mintzj** [comment]
I am going to try to ``rm(llist = ls())`` at the start of each vignette and see if that helps.

**Comment by mintzj** [comment]
That didn't do it.  It ran fine OK locally but not on Travis.

**Comment by ddalthorp** [comment]
I'm dubious...but it can't hurt!

On my machine locally, before the "error: ...ylim" crash, I get some other telling information:

```
* creating vignettes ...Warning: running command '"C:/PROGRA~1/R/R-34~1.2/bin/x64/Rscript" --vanilla --default-packages= -e "tools::buildVignettes(dir = '.', tangle = TRUE)"' had status 1
 ERROR
Warning in engine$weave(file, quiet = quiet, encoding = enc) :
  Pandoc (>= 1.12.3) and/or pandoc-citeproc not available. Falling back to R Markdown v1.
Warning in engine$weave(file, quiet = quiet, encoding = enc) :
  Pandoc (>= 1.12.3) and/or pandoc-citeproc not available. Falling back to R Markdown v1.
Quitting from lines 325-329 (wind-examples.Rmd) 
Error: processing vignette 'wind-examples.Rmd' failed with diagnostics:
need finite 'ylim' values
Execution halted
Error: Command failed (1)

```
I don't know precisely what to make of it, but it has to do with the building of tools to build vignettes. On my machine, it crashes the first time, but it does seem to get the vignette-building tools properly built. On the second pass, it works fine. On Travis, it's a cleaner start each time, so it might be like the first pass on my machine but it never does the second start.

**Comment by mintzj** [comment]
Usually things working on the second try for me indicates something to do with re-use of variable names or something else that is being re-used, like a plot that needs to have have a new plot created before it can be displayed.

**Comment by mintzj** [comment]
This is a really slippery bug.  On about eleven attempts to clean and rebuild the package then run ``devtools::build_vignettes()``, I get this error once.

**Comment by mintzj** [comment]
Got it again, without a fresh re-install.  Runs once, then fails to run! 
Restarting R session...

> library(GenEst)
> library("GenEst", lib.loc="~/R/win-library/3.4")
> detach("package:GenEst", unload=TRUE)
> devtools::build_vignettes()
Building GenEst vignettes
Moving command-line-example.html, solar-examples.html, wind-examples.html, command-line-example.R, solar-examples.R, wind-examples.R to inst/doc/
Copying command-line-example.Rmd, solar-examples.Rmd, wind-examples.Rmd to inst/doc/
> devtools::build_vignettes()
Building GenEst vignettes
Quitting from lines 362-370 (solar-examples.Rmd) 
Error: processing vignette 'solar-examples.Rmd' failed with diagnostics:
need finite 'ylim' values

**Comment by mintzj** [comment]
I think I will try to start over with a new pull request tomorrow.  I don't know what is going on here.

**Comment by DAPPERstats** [comment]
I'm heading home from practice now and can take a look at things. Is the
master Branch up to date?

**Comment by ddalthorp** [comment]
master branch looks good, I believe.

I pushed an update with expanded wind example vignette and everything passed, both in seeking-beta and subsequently in master

**Comment by DAPPERstats** [comment]
Sweet, thanks!

**Comment by DAPPERstats** [comment]
I'm running clean installations locally and getting vignette errors on a non-regular frequency.

what i do to test an installation is delete the existing instance I have of GenEst and then run the following (which works because i have Rtools set up):
```
devtools::install_github("ddalthorp/genest", build_vignettes = TRUE)
```
i have run this repeatedly (maybe like 20 times now?) to get a sense of reproducibility. 

once in every ~4 installation attempts, i get the `ylim` issue popping up, although it's not in a consistent spot:
```
Quitting from lines 645-649 (wind-examples.Rmd) 
Error: processing vignette 'wind-examples.Rmd' failed with diagnostics:
need finite 'ylim' values
```
```
Quitting from lines 325-329 (wind-examples.Rmd) 
Error: processing vignette 'wind-examples.Rmd' failed with diagnostics:
need finite 'ylim' values
```
```
Quitting from lines 461-466 (command-line-example.Rmd) 
Error: processing vignette 'command-line-example.Rmd' failed with diagnostics:
need finite 'ylim' values
```

the fact that this is happening with clean installation attempts means that it's not a result of old objects existing in the workspace and that it's not a result of vestigial from previous library builds.

rather, it is definitely something within the code written within the vignette that it does not always render a clean figure. 

each of those three spots is associated with a use of the function `calcSplits` 

i'm now digging into the vignettes to try and reproduce the failures

**Comment by ddalthorp** [comment]
I'm wondering if the `ylim` calculation in `plot.splitSummary` is a more likely spot for a bug than `calcSplits`

**Comment by DAPPERstats** [comment]
it's looking like `calcSplits` returns `NaN` values in some instances.

i have not been able to figure out why.

i can get the same inputs to `calcSplits` give me clean output or output with `NaN` values in a few spots. notably, they do not appear in the same spots when they do repeat. 

however, they always show up in pairs and always in a level corresponding to the last row of the summary table (for example "fall" in the season splits, even though they might be in any of the possible levels for CO splits if those exist, but always in the "fall" or in the last month ["200"] in the wind_RPbat example).

i have re-run the code about 50 times that's inside `calcSplits` outside of the function but using the same exact set of input values to the function that have given me NaN values, and not once has it given me an NaN.

basically, the output of `calcSplits` looks identical to the result of running the code inside the function except there are some `NaN` values.

i honestly have no idea what would cause this. if no one else has an idea of what the situation is, i think you can patch it temporarily by basically running the splits code multiple times and then taking a complete set. like i said, the `NaN`s don't come in the same spots (when they do come), so the likelihood of getting a duplicated `NaN` on even two runs seems vanishingly small. it's totally not a satisfying solution, but it should address the issue in the short term.

but in any case, at this point, i need to hand this off to @ddalthorp, as it looks like it's something with the split calculation functions.

**Comment by DAPPERstats** [comment]
nope, @ddalthorp, see my comment (we posted at nearly the exact same time!), signs point to it being in `calcSplits`

**Comment by ddalthorp** [comment]
strange.

I'll take a look.

**Comment by ddalthorp** [comment]
There's a very occasional issue in rcbinom in the tails of g. I'm fixing that now and will hit the `calcSplits` later today.

**Comment by mintzj** [comment]
Thank you so much, @DAPPERstats good work on tracing this down.  It is interesting to see how you approached searching for the cause.  My usual approach is take parts out until it works again, but that's not easy when the bug is intermittent.

**Comment by ddalthorp** [comment]
dang. I fixed a problem or two in qcbinom and ran the wind vignette a few hundred times without a problem. Uploaded to seeking-beta. It passed the tests just fine. I hit "merge" and now see the old ylim error.

**Comment by DAPPERstats** [comment]
that's quite likely a cache issue. i know it sounds weird, but because the vignettes use an existing build of the package (they have `library(GenEst)` calls), if there's an existing build of the package in the cache it will use that one

**Comment by mintzj** [comment]
I saw in the log that Travis re-installed cbinom when in ran through again.  I also got the error after installing cbinom 1.1 locally.  Do we have an idea of which plot is causing this?

**Comment by mintzj** [comment]
Seems like the line we get the error on really is just the last r block in the rmd file.

**Comment by DAPPERstats** [comment]
the issue is that the vignettes are built using the pre-existing version of genest

and it occurs in any/all of the split calculations.

**Comment by DAPPERstats** [comment]
no, in builds 769 and 768 it's failing on different lines in the same vignette

**Comment by mintzj** [comment]
Was successfully able to apply these changes in a separate branch, so closing this one now.
